src/main/java/Zap.java

```diff
@@ -0,0 +1,236 @@
+        System.out.println(" What can I do for you?");
+        System.out.println("____________________________________________________________");
+    }
+
+    /**
+     * main interaction loop of ZAP
+     * continuously prompts the user to enter commands and responds accordingly.
+     * The loop breaks when the user enters "thank you and bye."
+     * handles various commands like listing tasks,
+     * marking tasks as done, unmarking and adding new tasks
+     */
+    private static void processCommands() {
+        Scanner scanner = new Scanner(System.in);
+        String userCommand;
+
+        do {
+            System.out.print("Enter a command: ");
+            userCommand = scanner.nextLine();
+
+            if (userCommand.equalsIgnoreCase("thank you and bye")) {
+                break;
+            } else if (userCommand.equalsIgnoreCase("hi")) {
+                System.out.println("Hello! I am ZAP and I am at your service!");
+            } else if (userCommand.equalsIgnoreCase("bye")) {
+                System.out.println("You should say thank you, then say bye.");
+            } else if (userCommand.equalsIgnoreCase("list")) {
+                displayTasks();
+            } else if (userCommand.startsWith("mark")) {
+                markTask(userCommand);
+            } else if (userCommand.startsWith("unmark")) {
+                unmarkTask(userCommand);
+            } else if (userCommand.startsWith("todo")) {
+                addTodoTask(userCommand);
+            } else if (userCommand.startsWith("deadline")) {
+                addDeadline(userCommand);
+            } else if (userCommand.startsWith("event")) {
+                addEvent(userCommand);
+            } else {
+                System.out.println(userCommand);
+                addTask(userCommand);
+            }
+        } while (true);
+        scanner.close();
+    }
+
+    /**
+     * `addTask` method adds new task to the list of tasks
+     * creates a new `Task` object with description,
+     * adds it to the `tasks` list
+     */
+    private static void addTask(String taskDescription) {
+        if (taskDescription.isEmpty()) {
+            System.out.println("Woi. You think I robot then can waste my time >:(");
+        } else {
+            tasks.add(new Task(taskDescription));
+            System.out.println("____________________________________________________________");
+            System.out.println(" added: " + taskDescription);
+            System.out.println("____________________________________________________________");
+        }
+    }
+
+    private static void addTodoTask(String userCommand) {
+
+        String taskDescription = userCommand.substring(4).trim();
+        tasks.add(new TodoTask(taskDescription));
+        System.out.println("____________________________________________________________");
+        System.out.println(" Got it. Zappy boy added this task:");
+        System.out.println("   [T][ ]  " + taskDescription);
```

Comment:
Would it be better to add a \t to add tabs instead of spaces at the start of the line.